src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -167,6 +174,24 @@ public IdamRolesMappingResponse buildIdamRoleMappings(List<ServiceRoleMapping> s
         }
     }
 
+    /**
+     * Prepare caseworker data to be published as a message to topic.
+     * @param caseWorkerData list containing caseworker data
+     */
+    @Override
+    public void publishCaseWorkerDataToTopic(List<CaseWorkerProfile> caseWorkerData) {
+        List<String> caseWorkerIds = caseWorkerData.stream()
+                .map(CaseWorkerProfile::getCaseWorkerId)
+                .collect(Collectors.toUnmodifiableList());
+
+        ListUtils.partition(caseWorkerIds, caseWorkerDataPerMessage)
+                .forEach(data -> {
+                    PublishCaseWorkerData publishCaseWorkerData = new PublishCaseWorkerData();
```

Comment:
Can we initialise this outside the loop?

Reply:
sure